src/lib/layout/aem-model-provider/aem-model-provider.component.ts

```diff
@@ -46,12 +56,25 @@ export class AEMModelProviderComponent {
    */
   updateItem(): void {
     ModelManager.getData({ path: this.cqPath }).then(model => {
+      this.cqItemLoaded = Promise.resolve(true);
       this.cqItem = model;
-      this.aemComponent.changeDetectorRef.markForCheck();
+      if (this.pagePath && Utils.isInEditor()) {
+        PathUtils.dispatchGlobalCustomEvent(Constants.ASYNC_CONTENT_LOADED_EVENT, {});
```

Comment:
Just curious, what would happen if we have two components triggering these events in different times? Why are we doing content loaded on every updateItem and we will never do markForCheck? Is this function just ran once?

Reply:
That is indeed the intent. The logic here is purely concerning AEM in that every time model for a component is fetched/updated, the event is dispatched to let AEM know that the editable needs to be reloaded to use the new data.